plugins/catalog-react/src/components/UserListPicker/UserListPicker.tsx

```diff
@@ -180,23 +180,16 @@ export const UserListPicker = (props: UserListPickerProps) => {
   // external updates to the page location.
   useEffect(() => {
     if (queryParamUserFilter) {
-      setSelectedUserFilter(queryParamUserFilter as UserListFilterKind);
+      if (['group', 'user'].some(kind => kind === kindParameter)) {
+        setSelectedUserFilter('all');
+      } else {
+        setSelectedUserFilter(queryParamUserFilter as UserListFilterKind);
```

Comment:
This removed the `filterCounts[selectedUserFilter] === 0` check, was that intentional?

Reply:
@freben Yes that is intentional. As per the request the the selectedUserFilter should only automatically overide to `all` when the kind filter is `User` or `Group` for all other kinds the end user should be able to see if they dont have any entities that match `starred` or `owned`.

Reply:
are we sure of this approach? Have we evaluated if it's possible to strip out the filter in `EntityListProvider` before api calling to catalog instead?